utils/s2n_init.c

```diff
@@ -139,13 +139,7 @@ int s2n_cleanup(void)
 {
     POSIX_GUARD(s2n_cleanup_thread());
 
-    /* If this is the main thread and atexit cleanup is disabled,
-     * perform final cleanup now */
-    if (pthread_equal(pthread_self(), main_thread) && !atexit_cleanup) {
-        POSIX_GUARD(s2n_cleanup_final());
-    }
-
-    return 0;
+    return S2N_SUCCESS;
```

Comment:
I know what you mean, but s2n_cleanup only does a full cleanup when the atexit handler is disabled. So I think that test you've found was actually written incorrectly in the first place.

Reply:
Ah true - this test does look incorrect. But I guess I'm wondering, if this test *was* written correctly and called `s2n_disable_atexit()` before calling `s2n_cleanup()`, would this change have broken this test? If not, are there potentially other tests that call `s2n_cleanup()` that are written correctly, which may now not be testing what they should be?

Reply:
I don't think so. The thing is, we don't really test s2n_cleanup that much. When we call that function directly in tests, it's because we're trying to clean up everything. 
Additionally, as I mentioned, you only do a full cleanup in the tests when you call disable_atexit, and we basically never tested that function. So there's basically no tests that do a full cleanup anyways.

Reply:
Yeah that makes sense. I think that's generally true, but when I did a quick search I found a few suspicious cases. Like what about `s2n_random_rand_bytes_after_cleanup_cb`? It seems like this test might be invalid now:
https://github.com/aws/s2n-tls/blob/84561bbce17e1300693cb666962704a990203244/tests/unit/s2n_random_test.c#L785-L790

Reply:
This was resolved in your PR though, so maybe there are no more strange cases!